consensus/src/dag/rb_handler.rs

```diff
@@ -67,22 +72,24 @@ impl NodeBroadcastHandler {
         self.storage.delete_votes(to_delete)
     }
 
-    fn validate(&self, node: &Node) -> anyhow::Result<()> {
+    fn validate(&self, node: Node) -> anyhow::Result<Node> {
         let current_round = node.metadata().round();
 
         // round 0 is a special case and does not require any parents
         if current_round == 0 {
-            return Ok(());
+            bail!(NodeBroadcastHandleError::InvalidRound);
         }
 
         let prev_round = current_round - 1;
 
         let dag_reader = self.dag.read();
         // check if the parent round is missing in the DAG
-        ensure!(
-            prev_round >= dag_reader.lowest_round(),
-            NodeBroadcastHandleError::MissingParents
-        );
+        if prev_round >= dag_reader.lowest_round() {
+            if let Err(err) = self.fetch_requester.request_for_node(node) {
```

Comment:
I think we should at least unify this condition with the one below, it also doesn't look correct to me, maybe what you meant to check is prev_round > `highest_round`?

previously the check is to filter out node that below lowest round, the error was misleading

Reply:
Good catch. I did not account for check inversion - I was doing ensure! before. But, I removed this condition and just use the one with missing parents check below. Previously I was not checking for signatures if none of the parents existed locally, but I should have.